backend/app.py

```diff
@@ -44,86 +25,70 @@
-    prompt = f"""### System:\n{system_prompt}\n\n"""
-    
-    #for pair in hist:
-    #   prompt += f"""### User:\n{pair[0]}\n\n### Assistant:\n{pair[1]}\n\n"""
-
-    prompt += f"""### User:\n{user_inp}\n\n### Assistant:"""
+def prompt_build(system_prompt, user_input):
+    messages = [
+        {
+            "role": "system",
+            "content": system_prompt,
+        },
+        {"role": "user", "content": user_input},
+    ]
+    prompt = pipe.tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
     return prompt
 
-def chat(user_input):#, #history):
-    prompt = prompt_build(initial_system_prompt, user_input)#, history)
-    model_inputs = tokenizer([prompt], return_tensors="pt").to("cuda")
-
-    streamer = transformers.TextIteratorStreamer(tokenizer, timeout=10., skip_prompt=True, skip_special_tokens=True)
-
-    generate_kwargs = dict(
-        input_ids=model_inputs.input_ids,
-        attention_mask=model_inputs.attention_mask,
-        streamer=streamer,
-        max_length=8192,
-        do_sample=True,
-        top_p=0.95,
-        temperature=0.8,
-        top_k=50
-    )
-
-    thread = Thread(target=model.generate, kwargs=generate_kwargs)
-    thread.start()
-
-    model_output = ""
-    for new_text in streamer:
-        model_output += new_text
-    thread.join()  # Asegura que el hilo ha terminado
-
-    return model_output.strip()
+def clean_response(response, system_prompt, user_input):
+    # Remover el mensaje del sistema y cualquier marca sobrante
+    if system_prompt in response:
+        response = response.replace(system_prompt, "").strip()
+    if user_input in response:
+        response = response.replace(user_input, "").strip()
+    return response
+
+def chat(user_input):
+    prompt = prompt_build(initial_system_prompt, user_input)
+    outputs = pipe(prompt, max_new_tokens=1024, do_sample=True, temperature=0.7, top_k=50, top_p=0.95)
+    generated_text = outputs[0]["generated_text"].strip()
+    response = clean_response(generated_text, initial_system_prompt, user_input)
+    return response
 
 @app.route('/chat', methods=['POST'])
 def chat_api():
     data = request.json
-
-    if env_vars.verbose:
-        print(data)
-        
     user_input = data.get('user_input', '')
 
-    if env_vars.verbose:
-        print(user_input)
-    
-    # Cargar el contexto
+    # Cargar el contexto (historial)
     history = load_context()
 
-    response = chat(user_input)#, history)
+    response_1 = chat(user_input)
     
     # Actualizar el historial con la nueva interacción
-    #history.append((user_input, response))
+    # history.append((user_input, response))
     
     # Guardar el contexto actualizado
-    #save_context(history)
+    # save_context(history)
+
+    response_2 = clean_response_json(response_1)
     
-    return jsonify({"response": response})#, "history": history})
+    return jsonify({"response": response_2})
```

Comment:
To name this as "respose" or something like "filter_", "edited_"; something like that